tests/app/authentication/test_database_session_manager.py

```diff
@@ -0,0 +1,49 @@
+import unittest
+
+from mock import patch, Mock
+from sqlalchemy.exc import IntegrityError
+
+from app.authentication.session_management import DatabaseSessionManager, EQ_SESSION_ID
+
+
+class TestDatabaseSessionManager(unittest.TestCase):
+
+    @staticmethod
+    def test_store_user_id_rollback():
+        # Given
+        session_manager = DatabaseSessionManager()
+        user_id = "1"
+
+        with patch('app.authentication.session_management.session'), \
+                patch('app.authentication.session_management.db_session') as db_session:
```

Comment:
We should add `autospec=True` to the creation of the mock object here to ensure the assertions we make areactually testing what we are expeting them to (see https://engineeringblog.yelp.com/2015/02/assert_called_once-threat-or-menace.html)